### PR TITLE
Add Scylla Sphinx Theme 1.0

### PR DESCRIPTION
**Pull request by dgarcia360**
Replaces https://github.com/scylladb/scylla-operator/pull/802

Sphinx ScyllaDB Theme 1.0 is now released 🥳 

We’ve made a number of updates to the look and feel of the theme to improve the overall user experience.
You can read more about all notable changes [here](https://sphinx-theme.scylladb.com/stable/CHANGELOG#september-2021).

This PR also cleans the file ``conf.py``, removing several unsued options.

## How to test this PR

1. Clone this PR. For more information, see [Cloning pull requests locally](https://docs.github.com/en/github/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/checking-out-pull-requests-locally).

2. Enter the docs folder, and run:

```
make preview
````

3. Open http://127.0.0.1:5500/ with your favorite browser. You will see the docs with the new look and feel:

image

### CODE REVIEW

**Comment by tzach** [comment]
@dgarcia360 @tnozicka what is missing to merge this PR?
We want to move to the new theme

File: .github/workflows/docs-pages@v2.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+name: "Docs / Publish"
```
[the remark this answers is not in the document]
Comment (tnozicka) [reply .github/workflows/docs-pages@v2.yaml:1]: This is still outstanding. We have a version control system, let's use it.

File: .github/workflows/docs-pages@v2.yaml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+on:
+  push:
+    branches:
+    - master
+    paths:
+    - "docs/**"
+  workflow_dispatch:
```
[the remark this answers is not in the document]
Comment (tnozicka) [reply .github/workflows/docs-pages@v2.yaml:9]: ok, I think I see where the confusion is coming from. Conventionally, we pin all of our dependencies which means we always build the same output on re-trigger. I've noticed our docs don't, which is why it needs "workflow dispatch" hack. If we were to pin all the dependencies to make reproducible builds (and I think we should), you'd update by sending a PR and bumping that version without needing this artificial step, also make sure the dependency update passes CI. If deps can change randomly on us our CI can get borked at any time by an external change which is bad.

File: .github/workflows/docs-pr@v1.yaml [hunk 3]
```diff
@@ -0,0 +1,24 @@
+name: "Docs / Build PR"
```
[the remark this answers is not in the document]
Comment (tnozicka) [reply .github/workflows/docs-pr@v1.yaml:1]: if a job is required to pass on a PR, why shouldn't it be required to pass on master / after the merge?

File: docs/pyproject.toml [hunk 4]
```diff
@@ -9,7 +9,7 @@ python = "^3.7"
 pyyaml = "5.3"
 pygments = "2.2.0"
 recommonmark = "0.5.0"
-sphinx-scylladb-theme = "~0.1.11"
+sphinx-scylladb-theme = "~1.0.0"
```
Comment (tnozicka) [inline docs/pyproject.toml:12]: I think here (and everywhere else) we should pin with equality for the reasons mentioned above

Comment (dgarcia360) [reply docs/pyproject.toml:12]: "~" means update only if there is a new patch.

This allows us to release new theme versions with nonbreaking changes without having to submit a PR per repository.

If the major or minor is updated, the theme is not updated automatically.

More information: https://github.com/scylladb/sphinx-scylladb-theme/issues/75

File: .github/workflows/docs-pages@v2.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (dgarcia360) [reply .github/workflows/docs-pages@v2.yaml:9]: See https://github.com/scylladb/scylla-operator/pull/815#discussion_r760085861

File: .github/workflows/docs-pr@v1.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (dgarcia360) [reply .github/workflows/docs-pr@v1.yaml]: It *could* also run after the merge.

I've open this issue to run  ``make tests`` on  ``master`` branch *after* merge: https://github.com/scylladb/sphinx-scylladb-theme/issues/254

File: docs/pyproject.toml (hunk 4 above)
Comment (tnozicka) [reply docs/pyproject.toml:12]: I still believe we need to pin the dependencies but we could leave this and just add the poetry-lock file which solve this.

One of the goals of the CI run is reproducibility which would be negated by running with different dependency versions.

We needed to fix the python deps and while adding CI for it I had to pick most of the pieces from this PR in #886

**Pull request closed without merging**